### PR TITLE
Migrate older MeshModel endpoints to proposed MeshModel endpoints

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: Ashish Tiwari <ashishjaitiwari15112000@gmail.com>

**Notes for Reviewers**
Refer to https://docs.google.com/spreadsheets/d/1ABJCbfQRi0uN_YoP2kmHZ-lTI4S4QuvRk1_unomKNRE/edit#gid=1241442399 for proposed meshmodel endpoints

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Review by leecalcote — CHANGES_REQUESTED** [review]
Include API docs update.

**Comment by l5io** [comment]
🚀 Preview for commit 420ec5d39fd628de5c1e4d8dc514b82614661bff at: https://63beaad4b95e9519cd5a3acf--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 508d60de9a5d49d70bba75732b1bd1a3bd7f273a at: https://63bef638c893240a8f3169f2--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 791ea9fba31fb369c502b49145f15dfac1a41897 at: https://63bef80c08e2a70c06ade990--meshery-docs-preview.netlify.app

File: server/handlers/component_handler.go [hunk 1]
```diff
@@ -84,14 +84,14 @@ func (h *Handler) GetAllComponents(rw http.ResponseWriter, r *http.Request) {
 
 // swagger:route GET /api/meshmodel/model GetMeshmodelModels idGetMeshmodelModels
 // Handle GET request for getting all meshmodel models. The component type/model name should be lowercase like "kubernetes", "istio"
-// responses:
 // ?version={version}
 // ?order={field} orders on the passed field
 // ?category={category of model}
```
Comment (leecalcote) [inline server/handlers/component_handler.go:89]: What is the behavior in the absence of a value? Is there a default value?
Is this API greedy? Are partial matches supported?


Please answer these questions across each operand.

**Review by leecalcote — CHANGES_REQUESTED** [review]

**Comment by l5io** [comment]
🚀 Preview for commit dbf5276225c70b2c19ea2ed5c8f3ab7617baad61 at: https://63befb6779198d0da199988c--meshery-docs-preview.netlify.app

File: server/handlers/component_handler.go (hunk 1 above)
Comment (leecalcote) [reply server/handlers/component_handler.go:89]: Good. And now for these answers on every other operand.

**Comment by l5io** [comment]
🚀 Preview for commit 786337a47eda6ee7f7ca9a894ab084e58b08aec6 at: https://63befd89bcacda119e239f7e--meshery-docs-preview.netlify.app

File: server/handlers/component_handler.go (hunk 1 above)
Comment (Revolyssup) [reply server/handlers/component_handler.go:89]: The same is true for all operands

**Comment by l5io** [comment]
🚀 Preview for commit d7b0f481cf3ebdb08c6945bda4d720e8c4d8924b at: https://63bf01c308e2a71543ade87f--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit d490af7bca9f0eadc1c8b4748c7b3d1fc0f983db at: https://63bf1bae423d1e2eca9f2aae--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
I'll reconfirm: "Migrate" as a word used in the PR subject line isn't strictly the behavior of this PR in that meshmodel and oam will be offered concurrently, correct?

File: server/handlers/component_handler.go [hunk 2]
```diff
@@ -79,49 +80,94 @@ func (h *Handler) GetAllComponents(rw http.ResponseWriter, r *http.Request) {
...
+
+// swagger:route GET /api/meshmodel/model GetMeshmodelModels idGetMeshmodelModels
+// Handle GET request for getting all meshmodel models. The component type/model name should be lowercase like "kubernetes", "istio"
+// ?version={version} If version is unspecified then all models are returned
+// ?order={field} orders on the passed field
+// ?category={category of model} If category in unspecified then all models are returned
+// ?sort={[asc/desc]} Default behavior is asc
+// ?search={[true/false]} If search is true then a greedy search is performed
+// ?page={page-number} Default page number is 1
+// ?pagesize={pagesize} Default pagesize is 25
```
Comment (leecalcote) [inline server/handlers/component_handler.go:94]: @Revolyssup is there a value (e.g. "all" or "0") that is supported for unlimited pagesize?

File: server/handlers/component_handler.go [hunk 3]
```diff
@@ -79,49 +80,94 @@ func (h *Handler) GetAllComponents(rw http.ResponseWriter, r *http.Request) {
...
-// swagger:route GET /api/meshmodel/components MeshmodelGet idMeshmodelGet
-// Handle GET request for getting all meshmodel components.
-// Components can be further filtered through query parameter ?version=
+/**Meshmodel endpoints **/
+const DefaultPageSizeForMeshModelComponents = 25
+
+// swagger:route GET /api/meshmodel/model GetMeshmodelModels idGetMeshmodelModels
+// Handle GET request for getting all meshmodel models. The component type/model name should be lowercase like "kubernetes", "istio"
+// ?version={version} If version is unspecified then all models are returned
+// ?order={field} orders on the passed field
```
Comment (leecalcote) [inline server/handlers/component_handler.go:89]: What's the difference between `order` and `sort`. I get what `sort` is, but it's not evident what `order` is here. Is this more like `filter` by type?

File: server/meshmodel/components/core/v1.0.0/Application.json [hunk 4]
```diff
@@ -0,0 +1,24 @@
+{
+	"kind": "Application",
```
Comment (leecalcote) [inline server/meshmodel/components/core/v1.0.0/Application.json:2]: Whoa... how'd this sneak in here? 😉

File: server/handlers/component_handler.go (hunk 3 above)
Comment (Revolyssup) [reply server/handlers/component_handler.go:89]: Yes

**Comment by Revolyssup** [comment]
> I'll reconfirm: "Migrate" as a word used in the PR subject line isn't strictly the behavior of this PR in that meshmodel and oam will be offered concurrently, correct?

Yes

File: server/handlers/component_handler.go (hunk 2 above)
Comment (leecalcote) [reply server/handlers/component_handler.go:94]: In that case, let’s agree that “all” means return all results.

File: server/handlers/component_handler.go (hunk 3 above)
Comment (leecalcote) [reply server/handlers/component_handler.go:89]: Got it. Please update these docs to better explain the difference. 

Please toss in an example or two showing usage; e.g. how POSTs can be used.

File: server/meshmodel/components/core/v1.0.0/Application.json (hunk 4 above)
Comment (leecalcote) [reply server/meshmodel/components/core/v1.0.0/Application.json:2]: What happened to the other PR to unit this?

**Comment by leecalcote** [comment]
@theBeginner86 this PR impacts PR #6771 and is worth reviewing.

File: server/handlers/component_handler.go (hunk 1 above)
Comment (leecalcote) [reply server/handlers/component_handler.go]: Great. Clarify this in your docs. Consumers of these APIs shouldn't be left to guess each querystring option's behavior.

**Comment by leecalcote** [comment]
> > I'll reconfirm: "Migrate" as a word used in the PR subject line isn't strictly the behavior of this PR in that meshmodel and oam will be offered concurrently, correct?
> 
> Yes

Thanks for answering each of my questions. As you do, please incorporate your answer into the PR. In this case, that means: update the PR title to be accurate in its description of what this PR is actually doing.

File: server/handlers/component_handler.go (hunk 3 above)
Comment (Revolyssup) [reply server/handlers/component_handler.go:89]: model level POSTs are currently not supported as there are no registrants currently which register all entities combined at a model level

Comment (Revolyssup) [reply server/handlers/component_handler.go:89]: POSTs are at Entity level currently and their docs are mentioned on the comments of their handlers.

File: server/meshmodel/components/core/v1.0.0/Application.json (hunk 4 above)
Comment (Revolyssup) [reply server/meshmodel/components/core/v1.0.0/Application.json:2]: The component generator may have deleted that, I am checking logic right now to make sure that existing models are not deleted.

File: server/handlers/component_handler.go (hunk 2 above)
Comment (Revolyssup) [reply server/handlers/component_handler.go:94]: okay

Comment (Revolyssup) [reply server/handlers/component_handler.go]: Done

**Comment by l5io** [comment]
🚀 Preview for commit 66f8ee8a11f7d85272f940ae42c91434a2510e52 at: https://63bff2108a29d10062e5e403--meshery-docs-preview.netlify.app

**Pull request merged**